requirements.txt

```diff
@@ -1,5 +1,5 @@
 cryptography==3.4.8
-pyjwt
-requests
-boto3
-cachetools
+pyjwt==2.8.0
+requests==2.31.0
+boto3==1.34.15
+cachetools==5.3.2
```

Comment:
it's typically better to be a little flexible when pinning versions.  for example, `pyjwt~=2.8.0` will get all *compatible* versions 2.8.0 and higher.  i would suggest doing that for all libs in requirements.txt and requirements-dev.txt